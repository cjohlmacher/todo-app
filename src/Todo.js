import './Todo';
import Box from '@mui/material/Box';
import ListItem from '@mui/material/ListItem';
import ListItemText from '@mui/material/ListItemText';
import IconButton from '@mui/material/IconButton';
import DeleteOutlinedIcon from '@mui/icons-material/DeleteOutlined';

const Todo = ({description, deleteTodo}) => {
    const handleClick = () => {
        deleteTodo();
    };
    
    return (
        <Box sx={{width: '50%', bgcolor: '#322b2b', border: '1px solid white', margin: 'auto'}}>
            <ListItem>
                <ListItemText sx={{color: 'white'}} primary={description}></ListItemText>
                <IconButton aria-label="button" onClick={handleClick}>
                    <DeleteOutlinedIcon name="delete" sx={{color: 'white'}}/>
                </IconButton>
            </ListItem>
        </Box>
    );
};

export default Todo;