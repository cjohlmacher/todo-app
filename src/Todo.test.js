import { render } from '@testing-library/react';
import Todo from './Todo';

it('should render without breaking', () => {
    render(<Todo />);
});

it('should match the snapshot', () => {
    const {asFragment} = render(<Todo description="Wash dishes" deleteTodo={() => console.log('deleted')}/>);
    expect(asFragment()).toMatchSnapshot();
});