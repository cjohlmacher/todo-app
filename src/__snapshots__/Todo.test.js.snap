// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-361be1"
  >
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
    >
      <div
        class="MuiListItemText-root css-igjnpd-MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
        >
          Wash dishes
        </span>
      </div>
      <button
        aria-label="button"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1d159sf-MuiSvgIcon-root"
          data-testid="DeleteOutlinedIcon"
          focusable="false"
          name="delete"
          viewBox="0 0 24 24"
        >
          <path
            d="M16 9v10H8V9h8m-1.5-6h-5l-1 1H5v2h14V4h-3.5l-1-1zM18 7H6v12c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </li>
  </div>
</DocumentFragment>
`;
